yt_dlp/extractor/cda.py

```diff
@@ -173,18 +175,36 @@ def extract_format(page, version):
                     video['file'] = video['file'].replace('adc.mp4', '.mp4')
             elif not video['file'].startswith('http'):
                 video['file'] = decrypt_file(video['file'])
-            f = {
+            c_quality = video.get('quality')
+            qualities = video.get('qualities')
+            for q in qualities:
+                if qualities[q] == c_quality:
+                    c_quality = q
+            info_dict['formats'].append({
                 'url': video['file'],
-            }
-            m = re.search(
-                r'<a[^>]+data-quality="(?P<format_id>[^"]+)"[^>]+href="[^"]+"[^>]+class="[^"]*quality-btn-active[^"]*">(?P<height>[0-9]+)p',
-                page)
-            if m:
-                f.update({
-                    'format_id': m.group('format_id'),
-                    'height': int(m.group('height')),
-                })
-            info_dict['formats'].append(f)
+                'format_id': c_quality,
+                'height': int_or_none(c_quality[:-1]),
+            })
+            if qualities:
+                for quality in qualities:
+                    if quality != c_quality:
+                        data = {"jsonrpc": "2.0", "method": "videoGetLink",
+                                "params": [video_id, qualities[quality],
+                                           video.get('ts'), video.get('hash2'), {}], "id": 2}
+                        data = json.dumps(data).encode('utf-8')
+                        video_url = self._download_json('https://www.cda.pl/video/' + video_id, video_id,
+                                                        headers={'Content-Type': 'application/json',
+                                                                 'X-Requested-With': 'XMLHttpRequest'}, data=data,
+                                                        note=f"Fetching {quality} url",
+                                                        errnote=f"Failed to fetch {quality} url", fatal=False)
+                        if try_get(video_url, lambda x: x['result']['status']) == 'ok':
```

Comment:
again, confusing indenting

Reply:
I'm not sure what I can do about that one.